app/views/api/v2/taxonomies/show.json.rabl

```diff
@@ -6,52 +6,83 @@ attribute :ignore_types => :select_all_types
 
 attributes :description, :created_at, :updated_at
 
+ancestors = @taxonomy.ancestors.preload(
+  :users, :smart_proxies, :subnets, :compute_resources, :media, :ptables,
+  :provisioning_templates, :domains, :realms, :environments, :hostgroups
+)
 node :hosts_count do |taxonomy|
   hosts_count[taxonomy]
 end
 
-child :users do
+child (@taxonomy.users + ancestors.map(&:users).flatten).uniq => :users do
   extends "api/v2/users/base"
+  node :inherited do |current_user|
+    !@taxonomy.users.map { |user| user.id }.include?(current_user.id)
```

Comment:
Small nit: ActiveRecord implements `==` operator to basically `right.insatance_of?(self.class) && left.id == right.id` and we do not customize it anywhere I believe, we could thus reduce this with none performance impact to more readable
```suggestion
    !@taxonomy.users.include?(user)
```

Reply:
done :)